src/server-core/src/lib.rs

```diff
@@ -229,6 +230,50 @@ impl TlsCertConfig {
         builder.set_private_key_file(&self.key, SslFiletype::PEM)?;
         Ok(builder.build().into_context())
     }
+
+    /// Like [Self::context] but attempts to reload the files each time `ticker` yields an item.
+    /// Returns an error based on the files currently on disk. When `ticker` receives, the
+    /// certificates are reloaded from the context. The result of the reloading is returned on the
+    /// oneshot, and an Ok result means new connections will use the new certificates. An Err result
+    /// will not change the current certificates.
+    pub fn reloading_context(
+        &self,
+        mut ticker: BoxStream<'static, oneshot::Sender<Result<(), anyhow::Error>>>,
```

Comment:
I get the intention of the `Stream`, but since nothing takes action on the errors right now, what do you think about callers providing just a `Duration` that internally this method uses to create an `Interval`?

I don't feel super strong either way, so up to you!

Reply:
Makes sense!